records/generator_test.go

```diff
@@ -208,6 +210,14 @@ func TestInsertState(t *testing.T) {
 		{rg.As, "slave.mesos.", []string{"1.2.3.10", "1.2.3.11", "1.2.3.12"}},
 		{rg.As, "some-box.chronoswithaspaceandmixe.mesos.", []string{"1.2.3.11"}}, // ensure we translate the framework name as well
 		{rg.As, "marathon.mesos.", []string{"1.2.3.11"}},
+		{rg.SRVs, "_big-dog._tcp.marathon.mesos.", []string{
+			"big-dog-4dfjd-0.marathon.mesos.:80",
+			"big-dog-4dfjd-0.marathon.mesos.:443",
+		}},
+		{rg.SRVs, "_big-dog2._tcp.marathon.mesos.", []string{
```

Comment:
I think this is the (unintended?) behavior change. the "LEGACY" block you added causes this to be created. Prior to this PR the `big-dog2` task name is replaced by the `big-dog` discovery info name.

What's the justification for this change in behavior?